index.bs

```diff
@@ -575,79 +575,225 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
+				constructor(record&lt;DOMString, ClipboardItemData> items,
+				            optional ClipboardItemOptions options = {});
+
+				readonly attribute PresentationStyle presentationStyle;
+				readonly attribute FrozenArray&lt;DOMString> types;
+
+				Promise&lt;Blob&gt; getType(DOMString type);
+			};
 
-		typedef (DOMString or Blob) ClipboardItemDataType;
-		typedef Promise&lt;ClipboardItemDataType> ClipboardItemData;
+			enum PresentationStyle { "unspecified", "inline", "attachment" };
 
-		callback ClipboardItemDelayedCallback = ClipboardItemData ();
+			dictionary ClipboardItemOptions {
+				PresentationStyle presentationStyle = "unspecified";
+			};
+		</pre>
 
-		[Exposed=Window] interface ClipboardItem {
-		  constructor(record&lt;DOMString, ClipboardItemData> items,
-		    optional ClipboardItemOptions options = {});
-		  static ClipboardItem createDelayed(
-		      record&lt;DOMString, ClipboardItemDelayedCallback> items,
-		      optional ClipboardItemOptions options = {});
+		<dl class=note>
+			<dt><code><var>clipboardItem</var> = new ClipboardItem([<var>items</var>, <var>options</var>])</code>
+			<dd>
+			Creates a new {{ClipboardItem}} object. <var>items</var> are used to fill its MIME types and [=Promise=]s to [=Blob=]s or {{DOMString}}s corresponding to the MIME types, <var>options</var> can be used to fill its {{ClipboardItemOptions}},
+			as per the example below.
 
-		  readonly attribute PresentationStyle presentationStyle;
-		  readonly attribute long long lastModified;
-		  readonly attribute boolean delayed;
+				<pre class="example javascript" highlight=js>
+					const format1 = 'text/plain';
+					const promise_text_blob = Promise.resolve(new Blob(['hello'], {type: format1}));
+					const clipboardItemInput = new ClipboardItem(
+						{[format1]: promise_text_blob},
+						{presentationStyle: "unspecified"});
+				</pre>
 
-		  readonly attribute FrozenArray&lt;DOMString> types;
+			<dt><code><var>clipboardItem</var>.getType(<var>type</var>)</code>
+			<dd><p>Returns a [=Promise=] to the [=Blob=] corresponding to <var>type</var>.</p>
 
-		  Promise&lt;Blob> getType(DOMString type);
-		};
+			<dt><code><var>clipboardItem</var>.<var>types</var></code>
+			<dd><p>Returns the list of <var>types</var> contained in the <var>clipboardItem</var> object.
+
+		</dl>
+
+		<h4 id="clipboard-item">Clipboard Item</h4>
+
+		A [=clipboard item=] is conceptually data that the user has expressed a desire to make shareable by invoking a "cut" or "copy" command.
+
+		<p class=note>
+			For example, if a user copies a range of cells from a spreadsheet of a native application, it will result in one [=clipboard item=]. If a user copies a set of files from their desktop, that list of files will be represented by multiple [=clipboard item=]s.
+		</p>
+
+		Some platforms may support having more than one [=clipboard item=] at a time on the [=Clipboard=], while other platforms replace the previous [=clipboard item=] with the new one. 
```

Comment:
This sentence should be deleted. This section is not about multiple clipboard items, it's about one item. Section 7.3 is the correct section for this and it already contains that information.

Reply:
It also talks about the platforms that don't support multiple `Clipboard Item`s like Windows. I think the model should describe both single and multiple `Clipboard Item`/s. That way it is clear how implementers would implement the conceptual `Clipboard Item` object. Section 7.3 just describes the conceptual `Clipboard` object and the methods associated with it, not the `Clipboard Item`(single or multiple). The methods in `Clipboard` object use `Clipboard Items`, but `Clipboard Items` isn't part of `Clipboard` object. I did define the conceptual `Clipboard Items` there as that is used by the conceptual `Clipboard` object, but I can move that to a separate section if that would be better?

Reply:
Thanks for the response.

> It also talks about the platforms that don't support multiple Clipboard Items like Windows. 

That information is beyond the scope of _a_ Clipboard Item.

> I think the model should describe both single and multiple Clipboard Item/s.

Which model?

> That way it is clear how implementers would implement the conceptual Clipboard Item object.

That information is unnecessary for implementing the conceptual clipboard item.

> Section 7.3 just describes the conceptual Clipboard object and the methods associated with it, not the Clipboard Item(single or multiple).

No. That section defines `ClipboardItems` in IDL: `typedef sequence<ClipboardItem> ClipboardItems;`.

> The methods in Clipboard object use Clipboard Items, but Clipboard Items isn't part of Clipboard object.

The `typedef` isn't part of the interface defined in IDL (I wonder if it could be).

> I did define the conceptual Clipboard Items there as that is used by the conceptual Clipboard object

The definition is in IDL, https://webidl.spec.whatwg.org/#idl-typedefs. The definition isn't conceptual, IIUC. `Clipboard` is defined in IDL, it's not conceptual. @snianu, @domenic, please correct me if I'm wrong.

Reply:
There are two terms defined, https://pr-preview.s3.amazonaws.com/w3c/clipboard-apis/pull/158.html#index-defined-here, "ClipboardItems" and "Clipboard Items", that seems wrong and to be part of the confusion. IIUC, only the term `ClipboardItems` should be defined, via the `typedef`.

Reply:
The case 1) of "ClipboardItems" and "Clipboard Items" differs from the case 2) of "presentationStyle", "presentation style" and "PresentationStyle. Note the difference: in case 2) there's no "Presentation Style". I'll add a comment to section 7.3 because that's where "Clipboard Items" is defined.